### DISCUSSION TITLE
[WinError 2] The system cannot find the file specified

### DISCUSSION THREAD

**Discussion by LBeal-INTERA** [phydrus/phydrus#36] [Compiling Fortran Code]
Hi there, 
I'm running Pydrus Example 2 with Python 3.7.3, and I'm getting an error on the line 
" rs = ml.simulate()":

________________________________________________________________________________________________
...
File "C:\...\anaconda3.7\lib\subprocess.py", line 1178, in _execute_child
    startupinfo)

FileNotFoundError: [WinError 2] The system cannot find the file specified**

________________________________________________________________________________________________

In searching for this error, is seems that perhaps the version of python and subprocess.py are not compatible. Any opinion on this? 

Thanks,
Lakin

**Comment by raoulcollenteur** [comment]
Hi Lakin,
Thanks for trying out Pydrus! Subprocess is called by ml.simulate() to run the Hydrus executable. Did you compile the Fortran code (you should have a hydrus.exe) and set the right path to the executable in this line?

`ml = ps.Model(exe_name=path_to_exe`

Setting this path to the executable should solve the problem!

Cheers, Raoul

**Comment by LBeal-INTERA** [comment]
Hi Raoul,
Thanks for the quick reply. 
We've tried compiling the Fortran code with the pydrus/source/hydrus.exe using minGW for windows 10, but the .exe did not work:
___________________________________________________________________________________________________________

$ ./make.bat

C:...\source>gfortran HYDRUS.FOR INPUT.FOR HYSTER.FOR MATERIAL.FOR OUTPUT.FOR SINK.FOR SOLUTE.FOR TEMPER.FOR TIME.FOR WATFLOW.FOR -o "C: ...\source\\hydrus.exe"
MATERIAL.FOR:656:72:

         PAUSE 'xMualem: h is equal to zero!'
                                                                        1
Warning: Deleted feature: PAUSE statement at (1)
MATERIAL.FOR:740:72:

       IF(FB*FA.GT.0.) PAUSE 'Root must be bracketed for ZBRENT.'
                                                                        1
Warning: Deleted feature: PAUSE statement at (1)
MATERIAL.FOR:796:72:

       PAUSE 'ZBRENT exceeding maximum iterations.'
                                                                        1
Warning: Deleted feature: PAUSE statement at (1)
MATERIAL.FOR:874:72:

       pause 'too many steps in qromb'
                                                                        1
Warning: Deleted feature: PAUSE statement at (1)
MATERIAL.FOR:930:72:

           if(den.eq.0.)pause 'failure in polint'
                                                                        1
Warning: Deleted feature: PAUSE statement at (1)
SOLUTE.FOR:2043:72:

       IF(FB*FA.GT.0.) PAUSE 'Root must be bracketed for ZBRENT1.'
                                                                        1
Warning: Deleted feature: PAUSE statement at (1)
SOLUTE.FOR:2099:72:

       PAUSE 'ZBRENT1 exceeding maximum iterations.'
                                                                        1
Warning: Deleted feature: PAUSE statement at (1) 

___________________________________________________________________________________________________________


Any thoughts on this error? 

Lakin

**Comment by raoulcollenteur** [comment]
Did it not create the hydrus.exe despite these warnings? I am on a Mac so can't help here. @mvremec  might be able to help?

Did you try using the pre-compiled executable ?https://github.com/raoulcollenteur/pydrus/blob/master/source/hydrus.exe

Raoul

**Comment by mvremec** [comment]
Hi Lakin,
Thanks for you comment on this and trying out Pydrus!
The error message you get while compiling the Fortran code should not be a problem, a hydrus.exe should anyway be created in the source folder.

Try to:
1. Set the right path to the created hydrus executable in the source folder
(try also with the pre-compiled executable)
`exe = "C:\\pydrus\\source\\hydrus.exe""`
or
2. Copy the hydrus executable to the working folder where example2.py is located.

Matevz
